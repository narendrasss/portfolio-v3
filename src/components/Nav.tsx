import React from 'react';
import Main from './Main';
import Header from './Header';
import Link from './base/Link';
import List from './base/List';

export interface NavProps {
  url: string;
  links: LinkTo[];
}

const Nav: React.FC<NavProps> = ({ url, links }) => (
  <Main as="nav" p={1} bg="black" color="white" justifyContent="center">
    <Header url={url} links={links} />
    <List items={links}>
      {({ name, url }) => (
        <Link
          key={[name, url].join()}
          to={url}
          color="white"
          fontSize={2}
          fontWeight={600}
        >
          {name.charAt(0).toUpperCase() + name.slice(1)}
        </Link>
      )}
    </List>
  </Main>
);

export default Nav;
