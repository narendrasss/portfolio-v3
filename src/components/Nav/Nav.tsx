import React from 'react';
import Helmet from 'react-helmet';
import Header from '../Header/Header';
import Main from '../Main/Main';
import NavButton from '../NavButton/NavButton';
import { ContentProps } from '../Content/Content';
import List from '../base/List';
import Link from '../base/Link';

const Nav: React.FC<ContentProps> = ({
  title,
  description,
  url,
  links,
  onNavClick,
  ...props
}) => (
  <Main
    data-testid="main"
    p={1}
    bg="black"
    color="white"
    justifyContent="center"
    {...props}
  >
    <Helmet
      title={title}
      meta={[{ name: 'description', content: description }]}
    />
    <Header url={url} links={links} />
    <NavButton onClick={onNavClick} isOpen={true} display={['block', 'none']} />
    <List items={links}>
      {({ name, url }) => (
        <Link
          key={[name, url].join()}
          to={url}
          color="white"
          fontSize={2}
          fontWeight={600}
        >
          {name.charAt(0).toUpperCase() + name.slice(1)}
        </Link>
      )}
    </List>
  </Main>
);

export default Nav;
