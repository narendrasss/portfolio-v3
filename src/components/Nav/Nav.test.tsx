import React from 'react';
import { render } from 'react-testing-library';
import Nav from './Nav';

describe('Nav', () => {
  const url = 'narendras.io';
  const links = [
    {
      name: 'about',
      url: '/about',
    },
    {
      name: 'blog',
      url: '/blog',
    },
  ];

  it('renders without crashing', () => {
    const { asFragment } = render(<Nav url={url} links={links} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
