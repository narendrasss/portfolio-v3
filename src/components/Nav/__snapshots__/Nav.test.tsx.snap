// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders without crashing 1`] = `
<DocumentFragment>
  <main
    class="sc-bdVaJa sc-bwzfXH sc-gZMcBi iWKbcB"
    color="white"
    data-testid="main"
  >
    <header
      class="sc-bdVaJa sc-bwzfXH sc-iwsKbI dUmjld"
      width="100%"
    >
      <a
        class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat cYLLdT"
        color="black"
        font-family="sans"
        font-size="0,1"
        font-weight="600"
        href="/home"
      >
        narendras
        <span
          class="sc-bdVaJa sc-htpNat fYauba"
          font-family="sans"
          font-size="0,1"
          font-weight="300"
        >
          .io
        </span>
      </a>
      <nav
        class="sc-bdVaJa sc-dnqmqq hjWSCW"
        display="none,block"
      >
        <a
          class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat dRQZBK"
          color="black"
          font-family="sans"
          font-size="0,1"
          href="/projects"
        >
          projects
        </a>
        <a
          class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat dRQZBK"
          color="black"
          font-family="sans"
          font-size="0,1"
          href="/about"
        >
          about
        </a>
        <a
          class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat dRQZBK"
          color="black"
          font-family="sans"
          font-size="0,1"
          href="/blog"
        >
          blog
        </a>
        <a
          class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat dRQZBK"
          color="black"
          font-family="sans"
          font-size="0,1"
          href="/resume"
        >
          resume
        </a>
      </nav>
    </header>
    <button
      class="sc-bdVaJa sc-gqjmRU sc-jTzLTM hzYros sc-dnqmqq kXPwVz"
      data-testid="button-close"
      display="block,none"
      type="button"
      width="4.5rem"
    >
      <div
        class="sc-bdVaJa sc-dnqmqq sc-fjdhpX sc-jzJRlG gJpFoh"
        width="4.5rem"
      />
      <div
        class="sc-bdVaJa sc-dnqmqq sc-fjdhpX sc-jzJRlG gJpFoh"
        width="4.5rem"
      />
    </button>
    <a
      class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat Aqqzh"
      color="white"
      font-family="sans"
      font-size="3"
      font-weight="600"
      href="/projects"
    >
      Projects
    </a>
    <a
      class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat Aqqzh"
      color="white"
      font-family="sans"
      font-size="3"
      font-weight="600"
      href="/about"
    >
      About
    </a>
    <a
      class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat Aqqzh"
      color="white"
      font-family="sans"
      font-size="3"
      font-weight="600"
      href="/blog"
    >
      Blog
    </a>
    <a
      class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat Aqqzh"
      color="white"
      font-family="sans"
      font-size="3"
      font-weight="600"
      href="/resume"
    >
      Resume
    </a>
  </main>
</DocumentFragment>
`;
